src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAGenerateAppConfig.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using NuGet.Frameworks;
+using System.Linq;
+using System.Xml.Linq;
+using Xunit;
+
+namespace Microsoft.NET.Build.Tasks.UnitTests
+{
+    public class GivenAGenerateAppConfig
+    {
+        [Fact]
+        public void It_creates_startup_and_supportedRuntime_nod_when_there_is_not_any()
+        {
+            var doc =
+                    new XDocument(
+                        new XDeclaration("1.0", "utf-8", "true"),
+                        new XElement("configuration"));
+
+            GenerateAppConfig.AddSupportedRuntimeToAppconfigFile(doc, "net452");
+
+            doc.Element("configuration")
+                .Elements("startup")
+                .Single().Elements()
+                .Should().Contain(e => e.Name.LocalName == "supportedRuntime");
+        }
+
+        [Fact]
+        public void It_creates_supportedRuntime_nod_when_there_is_startup()
+        {
+            var doc =
+                    new XDocument(
+                        new XDeclaration("1.0", "utf-8", "true"),
+                        new XElement("configuration", new XElement("startup")));
+
+            GenerateAppConfig.AddSupportedRuntimeToAppconfigFile(doc, "net452");
+
+            doc.Element("configuration")
+                .Elements("startup")
+                .Single().Elements()
+                .Should().Contain(e => e.Name.LocalName == "supportedRuntime");
+        }
+
+        [Fact]
+        public void It_does_not_change_supportedRuntime_nod_when_there_is_supportedRuntime()
+        {
+            var doc =
+                    new XDocument(
+                        new XDeclaration("1.0", "utf-8", "true"),
+                        new XElement("configuration",
+                            new XElement("startup",
+                                new XElement("supportedRuntime",
+                                    new XAttribute("version", "v4.0"),
+                                    new XAttribute("sku", ".NETFramework,Version=v4.7.2")))));
+
+            GenerateAppConfig.AddSupportedRuntimeToAppconfigFile(doc, "net461");
+
+            XElement supportedRuntime = doc.Element("configuration")
+                .Elements("startup")
+                .Single().Elements("supportedRuntime")
+                .Single();
+
+            supportedRuntime.Should().HaveAttribute("version", "v4.0");
+            supportedRuntime.Should().HaveAttribute("sku", ".NETFramework,Version=v4.7.2");
+        }
+
+        // intersection of https://docs.microsoft.com/en-us/nuget/reference/target-frameworks
+        // and https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/startup/supportedruntime-element#version
```

Comment:
Use this map is good enough? Also, `net30` is not a tfm in dotnet core. So, this is not a valid tfm